src/Persistance/RepositoryImplementations/SystemUserRepository.cs

```diff
@@ -411,18 +417,22 @@ private static ValueTask<SystemUser> ConvertFromReaderToSystemUser(NpgsqlDataRea
 		        sui.created,
                 sui.external_ref,
                 sui.systemuser_type,
-                sui.accesspackages
+                sui.accesspackages,
+                sui.sequence_no
 	        FROM business_application.system_user_profile sui 
                 JOIN business_application.system_register sr  
                 ON sui.system_internal_id = sr.system_internal_id
 	        WHERE sr.system_id = @system_id
-	            AND sui.is_deleted = false;
+	            AND sui.is_deleted = false
+                AND sui.sequence_no > @sequence_no
+            ORDER BY sui.sequence_no ASC
```

Comment:
This should have a LIMIT clause. You're reading all the `SystemUser`s with this code, then throwing away everything but 1 page in the service.